src/include/sof/lib/uuid.h

```diff
@@ -84,17 +90,48 @@ struct sof_uuid_entry {
 			 va, vb, vc,					\
 			 vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7)	\
 	__section(".static_uuids")					\
-	static const struct sof_uuid_entry uuid_name = {		\
+	static const struct sof_uuid_entry uuid_name ## _ldc = {	\
 		{.a = va, .b = vb, .c = vc,				\
 		 .d = {vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7}},	\
 		entity_name "\0"					\
 	}
 
+/** \brief Declares runtime UUID (aaaaaaaa-bbbb-cccc-d0d1-d2d3d4d5d6d7) and name.
+ *
+ * UUID value from variables declared with this macro are accessible in
+ * runtime code - to dereference use SOF_RT_UUID()
+ *
+ * \param entity_name Name of the object pinted by the software tools.
+ * \param uuid_name Uuid symbol name used with SOF_UUID() and SOF_RT_UUID().
+ * \param va aaaaaaaa value.
+ * \param vb bbbb value.
+ * \param vc cccc value.
+ * \param vd0 d0 value (note how d0 and d1 are grouped in formatted uuid)
+ * \param vd1 d1 value.
+ * \param vd2 d2 value.
+ * \param vd3 d3 value.
+ * \param vd4 d4 value.
+ * \param vd5 d5 value.
+ * \param vd6 d6 value.
+ * \param vd7 d7 value.
+ */
+#define DECLARE_SOF_RT_UUID(entity_name, uuid_name,			\
+			 va, vb, vc,					\
+			 vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7)	\
+	DECLARE_SOF_UUID(entity_name, uuid_name,			\
+			 va, vb, vc,					\
+			 vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7);	\
+	const struct sof_uuid uuid_name = {				\
+		.a = va, .b = vb, .c = vc,				\
+		.d = {vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7}		\
+	}
```

Comment:
So both 'uuid_name ## ldc' and 'uuid_name' sof_uuid struct will be located in section .static_uuids with your change? This sounds odd to me. Do you need to change the linker scripts also?

Reply:
no, only `uuid_name ## _ldc` will be in dedicated section, `uuid_name` should be in `.rodata` section

Reply:
OK, then it makes sense, I don't need the linker script for my UUID PR any more after this applied.

Reply:
Right, the whole content of yours first commit (_platform: move uuid_entries to RODATA part of the FW_) should be dropped and this PR  content should be used instead.